src/kit/DataGrid/custom-renderers/utils.ts

```diff
@@ -105,7 +107,30 @@ export function drawTextWithEllipsis(
   x: number,
   y: number,
   maxWidth: number,
-): void {
+): TextMetrics {
   const ellipsisText = truncate(ctx, text, x, maxWidth);
   ctx.fillText(ellipsisText, x, y);
+  return ctx.measureText(ellipsisText);
+}
+
+export function drawTypeBadge(
+  ctx: CanvasRenderingContext2D,
+  typeName: 'number' | 'text' | 'date' | 'array' | 'unspecified',
+  x: number,
+  y: number,
+): void {
+  const tagFont = '400 8px';
+  const backgroundColor = '#13569c';
+
+  ctx.beginPath();
+  ctx.font = tagFont;
+  ctx.fillStyle = backgroundColor;
+  ctx.strokeStyle = backgroundColor;
+  ctx.lineWidth = 1;
+  roundedRect(ctx, x, y - 10, measureTextCached(typeName, ctx, tagFont).width + 16, 18, 4);
+  ctx.stroke();
+  ctx.fill();
+  ctx.fillStyle = '#fff';
```

Comment:
can we use the data grid theme here? the colors should probably respond to the overall theme. i see in the theme there's a `textBubble` and `bgBubble` theme that we could use here [source](https://docs.grid.glideapps.com/api/common-types#theme).

Reply:
## light mode

<img width="735" alt="image" src="https://github.com/user-attachments/assets/36cc526e-752c-4858-b55a-9ec39f96d66e">


## dark mode

<img width="749" alt="image" src="https://github.com/user-attachments/assets/c624bf83-2c71-4a1c-9858-af9f9948bd4b">

what do you think? I think it's hard to look at the type badge. Also, should badge color be the same or similar to hew badge color? (default badge)

<img width="712" alt="image" src="https://github.com/user-attachments/assets/eeee0797-b113-474a-9a4f-d3c108d97538">

Reply:
I agree -- looking at the theme it appears we use the `textBubble` theme variable as the border of the tags for the tag cell. That said, the badge background color does change with the theme -- could we reproduce the logic in the badge component here?

in addition, note that the badge font weight is bold. that's definitely something we want to retain.

Reply:
### Dark Mode

<img width="529" alt="image" src="https://github.com/user-attachments/assets/0689bd03-3a6b-45a9-9196-a9115fd0f339">

### Light Mode

<img width="491" alt="image" src="https://github.com/user-attachments/assets/4ed8f3e2-a784-4df1-bfb4-b2ae4859e9e5">